internal/http/services/owncloud/ocdav/report.go

```diff
@@ -117,21 +119,31 @@ func (s *svc) doFilterFiles(w http.ResponseWriter, r *http.Request, ff *reportFi
 				statRes.Info.Path = parts[3]
 			}
 
-			infos = append(infos, statRes.Info)
+			resourceInfos = append(resourceInfos, statRes.Info)
 		}
 
-		responsesXML, err := s.multistatusResponse(ctx, &propfindXML{Prop: ff.Prop}, infos, namespace, nil, nil)
+	} else if ff.Rules.MyOfficeFiles {
+		currentUser := appctx.ContextMustGetUser(ctx)
+		var err error
+		resourceInfos, err = s.myOfficeFilesManager.ListMyOfficeFiles(ctx, currentUser)
 		if err != nil {
-			log.Error().Err(err).Msg("error formatting propfind")
+			log.Error().Err(err).Msg("error listing MyOfficeFiles")
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
-		w.Header().Set(HeaderDav, "1, 3, extended-mkcol")
-		w.Header().Set(HeaderContentType, "application/xml; charset=utf-8")
-		w.WriteHeader(http.StatusMultiStatus)
-		if _, err := w.Write([]byte(responsesXML)); err != nil {
-			log.Err(err).Msg("error writing response")
-		}
+	}
+
+	responsesXML, err := s.multistatusResponse(ctx, &propfindXML{Prop: ff.Prop}, resourceInfos, namespace, nil, nil)
```

Comment:
I understand that spliting this code out of the if is much better, since it's used in both cases. On the other hand, it makes it harded to revert, which is what we need. If we duplicate the code, then the next time we revert, it's just a matter of removing the if, even if smth changed in the meantime with the favs code. Other option would be to split this in a separate commit, in which you move this code outside of the if, and the other commit just adds the new else if.

Reply:
Ah yes, good point. I'm a fan of the second option :) I'll tackle the other issues first, and once everything works I'll split this out into a second commit that we can keep